### ISSUE TITLE
insertBehind failing

### ISSUE THREAD

**Issue by dwincort** [lspitzner/pqueue#16]
The `insertBehind` function doesn't always insert behind.  Consider the following example:
```
import Data.PQueue.Prio.Min (insertBehind, empty)
failure = insertBehind 1 4 $ insertBehind 1 3 $ insertBehind 1 2 $ insertBehind 1 1 empty
-- failure == fromAscList [(1,1),(1,4),(1,2),(1,3)]
```

**Comment by lspitzner** [comment]
Right, thanks for reporting. I think I see what is wrong, but I am not sure if a fix is possible or if it will retain the complexity guarantee, especially the amortized O(1). I am somewhat inclined to simply remove the function (again).

If the function is important to you (or other users) I can try to fix it, though.

**Comment by dwincort** [comment]
Thanks for the response!  It's definitely a feature we use, but we may be able to work around it if there is no fix.  Perhaps one idea could be to fix it but not provide any complexity guarantee -- sort of a "use at your own risk" function.  If the performance is altogether too bad for our needs, we can make the switch to using `insert` instead, and if you (or someone) come up with a better fix in the future, the library can be seamlessly updated.

**Comment by lspitzner** [comment]
I have spend some time considering this problem, without too much success. I have played a bit on paper with how the trees look when only ever inserting elements with the same key, and it seems that you need to bubble everything around for a proper `insertBehind`, i.e. touch every node of the structure (all forests, all binomtrees, ..). So in such a worst-case, doing things via `toList` and `fromList` would not only land you in the same complexity class, but they would perform the same.

For non-worst-cases it might be possible to improve things, but it is not even clear to me what conditions would land you in a better complexity class, and the implementation is not trivial either.

I also realized how foolish i was to ever expect the current implementation to work - sorry for that.

As you suggested, I plan to not remove the function, but replace it with something that certainly works, probably via `toList` and `fromList`. This would make it an ugly `O(n)`. (i'll probably make it a major version just to reflect that change.)

I may be wrong in my prognosis; perhaps i just miss some clever trick that could make it work more efficiently. But at this point my inner manager recommends to just enlarge the key so you can emulate the behind-part with just the regular `insert` - it seems to be the lowest-cost, acceptable performance, smallest risk (of being broken in some way) solution.

**Comment by lspitzner** [comment]
Sorry for the large delay.

I have decided to remove `insertBehind` from PQueue.Min/Max and replaced it with an O(n) version for PQueue.Prio.Min/Max.

This was done in commit 2b447b22c13bab0155a7d4bfbcd4fef8c0193d7e and is published as 1.4.1.1.
